site/_posts/2023-02-03-single-sign-on-laravel-fusionauth.md

```diff
@@ -0,0 +1,388 @@
+FUSIONAUTH_BASE_URL=http://localhost:9011
+
+# Replace http://localhost with the address for your PHP application, if necessary
+# We'll create the route for `/auth/callback` later##
+FUSIONAUTH_REDIRECT_URI=http://localhost/auth/callback
+```
+
+Now, we need to alter the service provider responsible for listening to events
+at `app/Providers/EventServiceProvider.php` so we can tell Laravel that it should call our class when there is an event
+for Socialite. To do this, change its `$listen` property to add the following entry to the array:
+
+```php
+protected $listen = [
+    \SocialiteProviders\Manager\SocialiteWasCalled::class => [
+        \SocialiteProviders\FusionAuth\FusionAuthExtendSocialite::class . '@handle',
+    ],
+];
+```
+
+> Note: if you already have something in the `$listen` array (like `Registered::class => [...]`), please keep it and
+> just add a new entry to the array
+
+#### Creating new fields for our User model
+
+Migrations are the code-based version control for your database. Instead of having to share SQL queries with your
+coworkers so they can create a table or add a column everytime there is a change in the database, you can write code
+to programatically do this when running a command.
+
+In Laravel, to create a new migration you just need to run the `artisan make:migration` and inform a meaningful name for
+it. In the next line, we'll create a migration called `add_fusionauth_fields_user_table`:
+
+```shell
+$ ./vendor/bin/sail artisan make:migration add_fusionauth_fields_user_table
+```
+
+Then, go to your `database/migrations` folder, where you'll find a file there with the name you created and prefixed by
+today's date (e.g `2023_02_03_123000_add_fusionauth_fields_user_table.php`).
+
+Let's edit that file so we can describe the changes needed in our database. We are going to add three new columns to
+the `users` table, `fusionauth_id`, `fusionauth_token` and `fusionauth_refresh_token`, and change the `password` column
+so it can be `NULL`.
+
+```php
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * When running this migration, we want to create three new columns and change the password so it can be NULL
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('users',  function (Blueprint $table) {
+            $table->string('fusionauth_id', 36)->unique();
+            $table->text('fusionauth_token');
+            $table->text('fusionauth_refresh_token')->nullable();
+            $table->string('password')->nullable()->change();
+        });
+    }
+
+    /**
+     * When rolling back this migration, we want to drop the added columns and revert the password to NOT NULL again
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('users',  function (Blueprint $table) {
```

Comment:
Is this in a github repository somewhere?

Reply:
Creating one at [FusionAuth/fusionauth-example-laravel-single-sign-on](https://github.com/FusionAuth/fusionauth-example-laravel-single-sign-on)